cpu/sam0_common/Makefile.include

```diff
@@ -1,11 +1,20 @@
 # Define the CPU family so we can differentiate between them in the code
 CFLAGS += -DCPU_FAM_$(shell echo $(CPU_FAM) | tr 'a-z-' 'A-Z_')
 
+# Set ROM and RAM lengths according to CPU model
+ifneq (,$(filter samd21g18a samd21j18a saml21j18a samr21g18a,$(CPU_MODEL)))
+  export ROM_LEN = 0x40000
+  export RAM_LEN = 0x8000
+endif
+
+ROM_START_ADDR ?= 0x00000000
```

Comment:
This could actually go in the cortex m makefile. These addresses are sensible defaults for any cortex m CPU and can be overridden in the CPU specific makefiles where they differ instead.

Reply:
Hmmm... As far as I understand the memory spaces are not defined by ARM in an explicit way, but it's up to the silicon vendors to allocate their ROM and RAM to any address.

In RIOT we have, for ROM, at least 3 examples:

  - cc2538 at 0x00200000
  - stm32 family at 0x08000000
  - kinetis, nordic and sam at 0x00000000

For RAM is even more varied:

  - cc2538 generally at 0x20000000, but for some models at 0x20004000
  - kinetis, stm32 and sam0 at 0x20000000
  - NXP at 0x10000000, some at 0x40000000
  - sam3x8e at 0x20070000

Thus, in my opinion use standard values seem dangerous.

Reply:
alright, I agree, it's generalized enough in this place.